src/Analyzers/Core/Analyzers/SimplifyInterpolation/Helpers.cs

```diff
@@ -72,11 +72,11 @@ private static IOperation Unwrap(IOperation expression)
             if (expression is IInvocationOperation { TargetMethod: { Name: nameof(ToString) } } invocation &&
                 HasNonImplicitInstance(invocation) &&
                 !syntaxFacts.IsBaseExpression(invocation.Instance.Syntax) &&
-                !invocation.Instance.Type.IsRefLikeType)
+                !invocation.Instance.Type!.IsRefLikeType)
             {
                 if (invocation.Arguments.Length == 1 &&
                     invocation.Arguments[0].Value is ILiteralOperation { ConstantValue: { HasValue: true, Value: string value } } literal &&
-                    invocation.SemanticModel.Compilation.GetTypeByMetadataName(typeof(System.IFormattable).FullName!) is { } systemIFormattable &&
+                    invocation.SemanticModel!.Compilation.GetTypeByMetadataName(typeof(System.IFormattable).FullName!) is { } systemIFormattable &&
```

Comment:
Do u anticipate lots `SemanticModel!.` uses in this series of changes? If so should we consider a model like the IDE has where we introduce say `RequiredSemanticModel`?

Reply:
I anticipate no more than what was required to get this to compile.